include/ensmallen_bits/cmaes/cmaparameters.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file cmaesparametes.hpp
+ * @author John Hoang
+ * @author Marcus Edel 
+ * 
+ * Definition of the Covariance Matrix Adaptation Evolution Strategy as proposed
+ * by N. Hansen et al. in "Completely Derandomized Self-Adaptation in Evolution
+ * Strategies".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+#define ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+
+namespace ens{
+  template<typename MatType>
+  class CMAparameters {
+    template<typename U> friend class CMAES;
+    public:
+      typedef typename MatType::elem_type ElemType;
+      typedef typename MatTypeTraits<MatType>::BaseMatType BaseMatType; 
+    // To do list: different parameters constructor for various improvements or
+    // usage or parameters tuning
+      CMAparameters() {}; // doing nothing
+      /**
+       * @brief Construct a new CMAparameters object
+       * @param dim Starting point dimension
+       * @param lambda The population(Offspring sampled in each step) size 
+       * @param negativeWeight The implementation of negative weight improvement, 1 means yes 0 means no
+       */
+      CMAparameters(
+        const size_t dim,
+        const size_t lambda = 0,
+        const size_t negativeWeight = 1) : 
+        dim(dim),
+        lambda(lambda),
+        negativeWeight(negativeWeight)
+      {
+        initialize_params();
+      };
+      void initialize_params(){
```

Comment:
This could go into a separate function, available for all other CMAES optimizer versions.